src/network/clientpackethandler.cpp

```diff
@@ -228,29 +227,23 @@ void Client::handleCommand_AccessDenied(NetworkPacket* pkt)
 	u8 denyCode;
 	*pkt >> denyCode;
 
-	if (denyCode == SERVER_ACCESSDENIED_SHUTDOWN ||
-			denyCode == SERVER_ACCESSDENIED_CRASH) {
+	if (pkt->getRemainingBytes() > 0)
 		*pkt >> m_access_denied_reason;
-		if (m_access_denied_reason.empty())
+
+	if (m_access_denied_reason.empty()) {
+		if (denyCode != SERVER_ACCESSDENIED_CUSTOM_STRING) {
 			m_access_denied_reason = gettext(accessDeniedStrings[denyCode]);
+		} else if (denyCode >= SERVER_ACCESSDENIED_MAX) {
+			m_access_denied_reason = "Unknown";
+		}
+	}
+
+	if (denyCode == SERVER_ACCESSDENIED_TOO_MANY_USERS) {
+		m_access_denied_reconnect = true;
+	} else if (pkt->getRemainingBytes() > 0) {
```

Comment:
any reason to keep this inconsistency?

Reply:
For backward compatibility. Servers don't seem to send proper reconnect requests when kicking players out because of too many users:

https://github.com/minetest/minetest/blob/master/src/network/serverpackethandler.cpp#L219-L229

Therefore, either the above code should be modified (which, combined with removing the inconsistency, breaks compatibility) or this should be kept.